server/game/cards/01_SOR/events/YoureMyOnlyHope.ts

```diff
@@ -0,0 +1,40 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { EventCard } from '../../../core/card/EventCard';
+import { CostAdjustType } from '../../../core/cost/CostAdjuster';
+
+export default class YoureMyOnlyHope extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: '3509161777',
+            internalName: 'youre-my-only-hope',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Look at the top card of your deck',
+            immediateEffect: AbilityHelper.immediateEffects.lookAt((context) => ({
+                target: context.source.controller.getTopCardOfDeck()
+            })),
+            then: (thenContext) => ({
```

Comment:
It works either way, but we can do a slight optimization here by using `ifYouDo: (ifYouDoContext)` instead of `then` in which case it will shortcut evaluating the stuff inside the `then` if the deck is empty because the lookAt didn't happen